### PR TITLE
Migrations

### PR DESCRIPTION
**Pull request by tomchop**
This PR introduces a migration framework for ArangoDB and includes an initial migration to update the database to version 2.

The migration framework allows for incremental updates to the database schema and data, ensuring that the database remains consistent across different versions of the application.

This initial migration ensures that a new field introduced in https://github.com/yeti-platform/yeti/pull/1175, `is_valid`, is populated on all database Obseravbles.

### CODE REVIEW

File: core/migrations/migration.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    def __init__(self):
+        self.connect_to_db()
+
+    def connect_to_db(self):
+        raise NotImplementedError
+
+    def update_db_version(self, version: int):
+        raise NotImplementedError
+
+    def migrate_to_latest(self, stop_at: int = None):
```
Comment (udgover) [inline core/migrations/migration.py:18]: Why don't rely on migration name which already includes a number?

File: core/migrations/arangodb.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+
+class ArangoMigrationManager(migration.MigrationManager):
+    DB_TYPE = "arangodb"
+
+    def connect_to_db(self):
+        self.db = ArangoDatabase()
+        self.db.connect()
+
+        system_coll = self.db.collection("system")
```
Comment (udgover) [inline core/migrations/arangodb.py:12]: Reminder for later: we will have to update to async job all collection calls.

**Review by udgover — APPROVED** [review]
Really cool and needed PR! I'm approving it even if I have a question regarding migration number handling. 

Should we also include a command in yetictl rather than running `poetry run python core/migrations/arangodb.py`?

This will also be useful to handle `stop_at` argument.

File: core/migrations/arangodb.py (hunk 2 above)
Comment (tomchop) [reply core/migrations/arangodb.py:12]: done.

File: core/migrations/migration.py (hunk 1 above)
Comment (tomchop) [reply core/migrations/migration.py]: The functions are arbitrarily named, so we wanna calculate them from the order they've been pushed in

**Pull request merged**